OpenRA.Mods.Common/Traits/DockClientManager.cs

```diff
@@ -326,25 +355,43 @@ public bool CanTarget(Actor self, in Target target, ref TargetModifiers modifier
 			if (target.Type != TargetType.Actor)
 				return false;
 
-			cursor = enterCursor;
-			IsQueued = modifiers.HasModifier(TargetModifiers.ForceQueue);
 			var forceEnter = modifiers.HasModifier(TargetModifiers.ForceMove);
+			IsQueued = modifiers.HasModifier(TargetModifiers.ForceQueue);
 			OrderID = forceEnter ? "ForceDock" : "Dock";
 
-			if (requireForceMove() && !forceEnter)
-				return false;
+			var context = new CanTargetContext { Target = target, IsQueued = IsQueued, ForceEnter = forceEnter };
+			var result = canTarget(context);
+			cursor = result.Cursor;
+			return result.CanTarget;
+		}
 
-			if (!canTarget(target.Actor, forceEnter, IsQueued))
-				return false;
+		public virtual bool IsQueued { get; protected set; }
+
+		public record CanTargetContext
```

Comment:
```suggestion
		public readonly record struct CanTargetContext
```

why do we need the extra layer of indirection?

Reply:
The goal was to unify `canTarget`, `useEnterCursor` and `requireForceMove` action delegates in `DockActorTargeter` and make that logic for the consumer of the `DockActorTargeter` more explicit. Since you wanted to make `canTarget` return cursor and also integrate logic for blocked cursor into it, it seemed like it would get very bloated and difficult to understand with another two action delegates.

So I chose to make a single point/place for determine the logic for targeting and cursor logic. Single input (`CanTargetContext`) and single output (`CanTargetResult`) object/record/struct. The logic is fully managed by the consumer (`IIssueOrder.Orders` in `DockActorTargeter` and `TransformsIntoDockClient`)

Reply:
I meant why are you using a class?

Reply:
Changed to `readonly recort struct`